src/typescript/index.d.ts

```diff
@@ -0,0 +1,430 @@
+        /**
+         * Data to be sent in the transaction
+         */
+        readonly data: string;
+    }
+
+    export interface ISetPrivacyGroupLockState {
+        /**
+         * Privacy group ID to lock / unlock
+         */
+        readonly privacyGroupId: string;
+        /**
+         * Private Key used to sign transaction with
+         */
+        readonly privateKey: string;
+        /**
+         * Orion public key
+         */
+        readonly enclaveKey: string;
+        /**
+         * boolean indicating whether to lock or unlock
+         */
+        readonly lock: boolean;
+    }
+
+    export interface IAddToPrivacyGroupOptions {
+        /**
+         * Privacy group ID to add to
+         */
+        readonly privacyGroupId: string;
+        /**
+         * Private Key used to sign transaction with
+         */
+        readonly privateKey: string;
+        /**
+         * Orion public key
+         */
+        readonly enclaveKey: string;
+        /**
+         * list of enclaveKey to pass to the contract to add to the group
+         */
+        readonly participants: string[];
+    }
+
+    export interface IRemoveFromPrivacyGroupOptions {
+        /**
+         * Privacy group ID to add to
+         */
+        readonly privacyGroupId: string;
+        /**
+         * Private Key used to sign transaction with
+         */
+        readonly privateKey: string;
+        /**
+         * Orion public key
+         */
+        readonly enclaveKey: string;
+        /**
+         * single enclaveKey to pass to the contract to add to the group
+         */
+        readonly participant: string;
+    }
+
+    export interface ICreatePrivacyGroupOptions {
+        /**
+         * Privacy group ID to add to
+         */
+        readonly privacyGroupId: string;
+        /**
+         * Private Key used to sign transaction with
+         */
+        readonly privateKey: string;
+        /**
+         * Orion public key
+         */
+        readonly enclaveKey: string;
+        /**
+         * list of enclaveKey to pass to the contract to add to the group
+         * Expected to be Base64 encoded array of strings.
+         */
+        readonly participants: string[];
+    }
+
+    export interface IFindOnChainPrivacyGroupOptions {
+        readonly addresses: string[];
+    }
+
+    export interface IGeneratePrivacyGroupOptions {
+        readonly privateFor: string[];
+        readonly privateFrom: string;
+    }
+
+    export interface IDeletePrivacyGroupOptions {
+        readonly privacyGroupId: string;
+    }
+
+    export interface IFindPrivacyGroupOptions {
+        readonly addresses: string[];
+    }
+
+    export interface IGetTransactionCountNoPrivacyGroupIdOptions {
+        readonly from: string;
+        readonly privateFrom: string;
+        readonly privateFor: string[];
+        readonly privacyGroupId?: string;
+    }
+    export interface IGetTransactionCountWithPrivacyGroupIdOptions {
```

Comment:
since this interface applies for the one with `privacyGroupId` better remove `privateFrom` and `privateFor` same applies for `IGetTransactionCountNoPrivacyGroupIdOptions`

Reply:
@achraf17 I'm not sure I get it, apologies. Could you elaborate?

If I remove `privateFrom` and `privateFor` from both of those interfaces then getTransactionCount will not be callable with those parameters specified, e.g. this becomes forbidden as far as the compiler is concerned (quoting index.js of web3-eea)

```js
web3.priv
      .getTransactionCount({
        from,
        privateFrom: options.privateFrom,
        privateFor: options.privateFor,
        privacyGroupId: options.privacyGroupId
      })
```

Reply:
I think this looks good as well. 

The quoting is valid as its only cascading the options param that was passed in to `genericSendRawTransaction`.